src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -505,7 +518,7 @@ def _validate_vm_create_storage_profile(cmd, namespace, for_scale_set=False):
             namespace.attach_data_disks = [_get_resource_id(cmd.cli_ctx, d, namespace.resource_group_name, 'disks',
                                                             'Microsoft.Compute') for d in namespace.attach_data_disks]
 
-    if not namespace.os_type:
+    if not namespace.os_type and namespace.storage_profile != StorageProfile.SharedGalleryImage:
```

Comment:
Is shared gallery image excluded in this logic?

Reply:
Yes, because the shared gallery image does not require `os_type`.